event-handlers/src/main/java/no/sikt/nva/nvi/events/evaluator/EvaluatorService.java

```diff
@@ -204,15 +194,36 @@ private boolean doesNotMeetNviRequirements(JsonNode publication) {
     return !isNviCandidate(model);
   }
 
-  private boolean existsAsCandidateInClosedPeriod(URI publicationId) {
+  private Optional<Candidate> fetchOptionalCandidate(URI publicationId) {
     try {
-      var existingCandidate =
+      return Optional.of(
           Candidate.fetchByPublicationId(
-              () -> publicationId, candidateRepository, periodRepository);
-      return Status.CLOSED_PERIOD.equals(existingCandidate.getPeriod().status());
+              () -> publicationId, candidateRepository, periodRepository));
     } catch (CandidateNotFoundException notFoundException) {
+      return Optional.empty();
+    }
+  }
+
+  /*
+   * Checks that the publication can be evaluated in the target period.
+   * If the period is closed, only existing candidates should be re-evaluated.
+   * New candidates can only be evaluated in an open period.
+   */
+  private boolean canEvaluateInNviPeriod(String year, Candidate optionalCandidate) {
```

Comment:
Send candidate to this methods, check if candidate is present in evaluateCandidacy method :)